modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionPhysicalProductPricer.java

```diff
@@ -137,13 +137,26 @@ public double impliedVolatility(
     ResolvedSwap underlying = swaption.getUnderlying();
     ResolvedSwapLeg fixedLeg = fixedLeg(underlying);
     ArgChecker.isTrue(expiry >= 0d, "Option must be before expiry to compute an implied volatility");
-    double forward = getSwapPricer().parRate(underlying, ratesProvider);
+    double forward = forwardRate(swaption, ratesProvider);
     double pvbp = getSwapPricer().getLegPricer().pvbp(fixedLeg, ratesProvider);
     double strike = getSwapPricer().getLegPricer().couponEquivalent(fixedLeg, ratesProvider, pvbp);
     double tenor = swaptionVolatilities.tenor(fixedLeg.getStartDate(), fixedLeg.getEndDate());
     return swaptionVolatilities.volatility(expiry, tenor, strike, forward);
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Provides the forward rate.
+   * This is the par rate for the forward starting swap that is the underlying of the swaption.
+   *
+   * @param swaption  the swaption
+   * @param ratesProvider  the rates provider
+   * @return the forward rate
+   */
+  public double forwardRate(ResolvedSwaption swaption, RatesProvider ratesProvider) {
+    return VolatilitySwaptionProductPricer.forwardRate(swaption, ratesProvider);
```

Comment:
Instead of calling the product pricer, you can directly call the swap pricer here. Same for the CashParYield pricer. Then you can do the same as in the product pricer for the other methods and call the method for the relevant pricer.